pkg/operator/ceph/controller/spec.go

```diff
@@ -670,15 +670,17 @@ func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Contai
 		maxLogSize = resource.MustParse(fmt.Sprintf("%dM", size))
 	}
 
-	periodicity := "daily"
-	if c.LogCollector.Periodicity == "1h" {
+	var periodicity string
+	if c.LogCollector.Periodicity == "1h" || c.LogCollector.Periodicity == "hourly" {
 		periodicity = "hourly"
-	} else if strings.Contains(c.LogCollector.Periodicity, "h") || strings.Contains(c.LogCollector.Periodicity, "d") {
-		periodicity = "daily"
-	} else if c.LogCollector.Periodicity != "" {
+	} else if c.LogCollector.Periodicity == "weekly" || c.LogCollector.Periodicity == "monthly" {
 		periodicity = c.LogCollector.Periodicity
+	} else {
```

Comment:
I think a better structure for this conditional statement would be:
```
if 1h or hourly
   hourly
else if 1d or 24h
  daily
else
 // Weekly or Monthly
  periodicity = c.LogCollector.Periodicity
```
This makes a bit more clear of what options the user can provide.

Reply:
to support the upgrade cluster, we have to support inputs like `2h`, 3h` or so... and by default, we want `periodicity` to `daily`, so I kept `daily` in the else block.